### PR TITLE
[steamworks-sdk] Add new port

### PR DESCRIPTION
**Pull request by BamButz**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [ ] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html).
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by BamButz** [comment]
Seems that the pipelines are trying to extract the zip file by using tar.
I will look into it again tomorrow, until that maybe someone got a hint for me :)

**Comment by dg0yt** [comment]
Not sure if this will be accepted, but...
 - CMake packages which are not from upstream must be prefixed with `unofficial-`.
   https://learn.microsoft.com/en-us/vcpkg/contributing/maintainer-guide#add-cmake-exports-in-an-unofficial--namespace
 - Add a single config package, not a module + wrapper.
 - You don't need four different versions of `find_library`. (It doesn't take the full filename.)
 - Consider providing an imported target with multi-config properties. (But I assume you are not familiar enough with CMake.)
 - There should be more comments, but it would be easier after getting rid of the redundant cmake stuff.

**Comment by BamButz** [comment]
Thanks for your feedback. I'll try to address the points accordingly.

File: ports/steamworks-sdk/usage [hunk 1]
```diff
@@ -0,0 +1,5 @@
+
+The package steamworks-sdk is compatible with CMake as following
+
+  find_package(Unoffical-Steamworks-SDK REQUIRED)
```
Comment (Cheney-W) [inline ports/steamworks-sdk/usage:4]:
```suggestion
  find_package(unofficial-steamworks-sdk REQUIRED)
```

File: ports/steamworks-sdk/vcpkg.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
+{
+  "name": "steamworks-sdk",
+  "version": "1.60",
+  "description": "Steamworks is a free collection of tools available to all developers for use in their games or software on Steam.",
+  "homepage": "https://partner.steamgames.com/doc/sdk",
+  "license": null,
+  "supports": "((windows & x64) | (windows & x86) | (osx & x64) | (linux & x64) | (linux & x86)) & !static & !uwp",
```
Comment (Cheney-W) [inline ports/steamworks-sdk/vcpkg.json:7]:
```suggestion
  "supports": "!static & !uwp & !arm",
```
Does this port support mingw and arm？

File: ports/steamworks-sdk/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,67 @@
+vcpkg_download_distfile(ARCHIVE
+    URLS "https://partner.steamgames.com/downloads/steamworks_sdk_160.zip"
+    FILENAME "steamworks_sdk_160.zip"
+    SKIP_SHA512 # Not sure why, but the SHA512 hash is changing with each download
+)
```
Comment (dg0yt) [inline ports/steamworks-sdk/portfile.cmake:5]: Hm, should this be accepted at all?
It cannot be verified and it cannot be cached.
IIUC the download links are hidden behind a registration when trying to find them on the website.

**Review by dg0yt — COMMENTED** [review]
There might be more comments, but:

File: ports/steamworks-sdk/portfile.cmake (hunk 3 above)
Comment (BamButz) [reply ports/steamworks-sdk/portfile.cmake]: I have to agree with you here, I mistakenly assumed that I had retrieved the link in incognito mode. This means that the archive probably cannot be retrieved in the correct way and the pull request should therefore be rejected.

**Comment by BamButz** [comment]
Closing this one because of https://github.com/microsoft/vcpkg/pull/41290#discussion_r1792583283

**Pull request closed without merging**